File: streaming/examples/enron/enron_map.py
#!/usr/bin/env python

import sys
sys.path.append(".")

from pymongo_hadoop import BSONMapper

def mapper(documents):
    i = 0
    for doc in documents:
        i = i + 1
        if 'headers' in doc and 'To' in doc['headers'] and 'From' in doc['headers']:
            from_field  = doc['headers']['From']
            to_field = doc['headers']['To']
            recips = [x.strip() for x in to_field.split(',')]
            for r in recips:
                yield {'_id': {'f':from_field, 't':r}, 'count': 1}

BSONMapper(mapper)
print >> sys.stderr, "Done Mapping."

File: streaming/examples/enron/enron_map.js
#!/usr/bin/env node

var node_mongo_hadoop = require('node_mongo_hadoop')


var trimString = function(str){
  return String(str).replace(/^\s+|\s+$/g, '');
}

function mapFunc(doc, callback){
    if(doc.headers && doc.headers.From && doc.headers.To){
        var from_field = doc['headers']['From']
        var to_field = doc['headers']['To']
        var recips = []
        to_field.split(',').forEach(function(to){
          callback( {'_id': {'f':from_field, 't':trimString(to)}, 'count': 1} )
        });
    }
}

node_mongo_hadoop.MapBSONStream(mapFunc);
